mne/io/nirx/nirx.py

```diff
@@ -133,15 +133,27 @@ def __init__(self, fname, saturated, preload=False, verbose=None):
                 if (key == 'wl1' or key == 'wl2') \
                     and len(glob.glob('%s/*%s' %
                                       (fname, 'nosatflags_' + key))) == 1:
+
+                    # Here two files have been found, one that is called
+                    # no sat flags. The nosatflag file has no NaNs in it.
+                    # The wlX file has NaNs in it.
+                    # First we determine which of the files has the saturation
+                    # flags in it.
+                    satidx = np.where(['nosatflags' not in op.basename(k)
+                                       for k in files[key]])[0][0]
+
                     if saturated == 'nan':
+                        # In this case data is returned with NaNs in it.
                         warn('The measurement contains saturated data. '
                              'Saturated values will be replaced by NaNs.')
-                        files[key] = files[key][1]
+                        files[key] = files[key][satidx]
```

Comment:
In addition to the test file issue, there was a bug in the original authors code. And I guess that's why we need tests :)